usr/share/rear/conf/default.conf

```diff
@@ -570,11 +570,12 @@ OPAL_PBA_USE_SERIAL_CONSOLE=""
 #   command is not issued for some reason, devices will remain unlocked permanently.
 OPAL_PBA_UNLOCK_MODE="transient"
 #
-# PBA debug password (empty for not using the debug shell facility).
-# If this pseudo-password is entered when the PBA asks for a password to unlock disks,
-# an interactive emergency shell will be started.
-# Used to debug the PBA system.
-OPAL_PBA_DEBUG_PASSWORD=""
+# PBA debug password as a salted SHA512 hash (empty for not using the debug shell facility).
+# If the debug password is entered when the PBA asks for a password to unlock disks,
+# an interactive emergency shell will be started, which can be used to debug the PBA system.
+# To generate a password hash, use the output of 'openssl passwd -6', copy its entire output
```

Comment:
On my openSUSE leap 15.1 system (which is like SLES15)
`openssl passwd -6` does not work
```
# echo "my_password" | openssl passwd -6 -stdin
passwd: Option unknown option -6
passwd: Use -help for summary.
```
but
```
# echo "my_password" | openssl passwd -1 -stdin
$1$36zeN549$H/fqeTN2RsH0mPQEIXMcW1
```
works, cf. what is currently described in default.conf for SSH_ROOT_PASSWORD
```
# To avoid a plain text password in the etc/rear/local.conf config file
# generate a hashed password with the command
#   echo "my_recovery_system_root_password" | openssl passwd -1 -stdin
# and use the output of openssl to set SSH_ROOT_PASSWORD='output_of_openssl'
# (single quotes avoid issues with the special bash character $ in the openssl output).
```
https://github.com/rear/rear/blob/master/usr/share/rear/conf/default.conf#L1556

Reply:
Thanks for checking this. Which openssl version is present on Leap 15.1?

`-1` uses the MD5 hashing algorithm with [known cryptographic weaknesses](https://en.wikipedia.org/wiki/MD5#Security), although the preimage resistance relevant here does not seem to be broken in practice yet. I also acknowledge that this is not that important on a PBA system: You'd have to tamper with the hardware (e.g. install another boot medium) in order to look into the PBA's file system in the first place and then you already have full control over the system (which still does not enable you to unlock primary disk contents). However, if possible, I'd like to avoid giving a suboptimal example for users of ReaR by suggesting a weak method to store password hashes.

Anyway, I'll do some research and come up with a solution which also works on (presumably) older openssl versions.

Reply:
It looks like Leap 15.1 comes with openssl 1.1.0, which only supports MD5 hashes (`-1`) and some exotic variants. The successor, openssl 1.1.1 also supports `-5` (SHA256) and `-6` (SHA512).

Reply:
On my openSUSE Leap 15.1 system I get
```
# openssl version
OpenSSL 1.1.0i-fips  14 Aug 2018
```

I think because ReaR is also used on SLES12 and some older RHEL versions
examples in ReaR comments or documentation may provide generally working things
plus perhaps an optional note that tells about known cryptographic weaknesses?

Reply:
a2f22c4ddec836dbbf3f29f1276f149abb571782 should address this as suggested.